src/Tools/DynamoInstallDetective/ProductLookUp.cs

```diff
@@ -159,7 +159,7 @@ static string GetDisplayName(RegistryKey key)
         public InstalledProductLookUp(string lookUpName, string fileLookup)
         {
             this.ProductLookUpName = lookUpName;
-            this.fileLocator = (path) => Directory.GetFiles(path, fileLookup).FirstOrDefault();
+            this.fileLocator = (path) => Directory.GetFiles(path, fileLookup, SearchOption.AllDirectories).FirstOrDefault();
```

Comment:
@aparajit-pratap Did you notice any noticeable slow down with such searching option with RSA?

Reply:
I didn't notice any difference as I don't suspect there is a noticeable slowdown if any after this change. In any case, I think this part of the fix might be unavoidable due to the fact that there is no guarantee that ASM will always be found in the root directory of an installation.

Reply:
@aparajit-pratap Thank you, sounds reasonable here.